# Group Increments Handout Discussion - Problem 1
After exploring the class exercise notebook, here is a notebook that contains my code for exploring the questions in tha handout. Everything is cut and paste from the class exercise notebook with changes as needed.

We begin with the following code will import the data and call the required libraries.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats

data_file = "Group_Increments.csv"    # This is the data file in the GitHub repository

############################
### Set global variables ###
############################

data_path = "https://raw.githubusercontent.com/blinkletter/4410PythonNotebooks/main/Class_05/data/"
image_path = "https://raw.githubusercontent.com/blinkletter/4410PythonNotebooks/main/Class_05/images/"

github_data_location = "https://raw.githubusercontent.com/blinkletter/LFER-QSAR/main/data/"
github_styles_location = "https://raw.githubusercontent.com/blinkletter/LFER-QSAR/main/styles/"



data_file = github_data_location + data_file

print(f"The data file name is {data_file}")

df1 = pd.read_csv(data_file,    # create pandas dataframe from data file
                  delimiter = ",",           # data is separated by commas
                  index_col="group",         # use the column with "group" as a header for the index
                  comment = "#",             # ignore lines in the file that are comments
                  skipinitialspace = True    # ignore extra spaces
                 ) 

#display(df1.head())     # display first five rows at head of Pandas dataframe
display(df1)

The data file name is https://raw.githubusercontent.com/blinkletter/LFER-QSAR/main/data/Group_Increments.csv


dHf
group           
C-H3C     -10.20
C-H2C2     -4.93
C-HC3      -1.90
C-C4        0.50
Cd-H2       6.26
...          ...
N-(CO)C2   14.40
N-(CO)CbH   0.40
N-(CO)2H  -18.50
N-(CO)2C   -5.90
N-(CO)2Cb  -0.50

[85 rows x 1 columns]

## Manipulating the Data
The data table is in units of kcal/mole. We usually use kJ/mole so let us convert the numbers to that unit and add a new column. We will then extract the new data series out of the dataframe. It will still be indexed by "group". let us grab just the series that is in kJ/mole and then we can access individual values using the index. Examine the following code...

In [2]:
n = 90      # num,ber of rows to display.  I want it all this time.
pd.set_option('display.max_rows', n) #replace n with the number of rows you want to see completely

df1["dHfkJ"] = df1["dHf"]*4.18        # add a series in units of Kj/mole
#display(df1)

i=df1["dHfkJ"]             # extract a series of group increments in kJ/mole as "i"
#display(i.head())

## Handout Problem 1
"Calculate the $\Delta H^{\circ}_f$ for methylcyclohexane, 1-methylcyclohexene and methylenecyclohexane. Compare your answers to experimental values. Is there any strain evident in these structures?"

See NIST website for experimental values at https://webbook.nist.gov/chemistry/

The code is all below and hopefull will explain itself. Observe how there are three similar blocks of code for each case where I just copied and pasted them and made changes as needed.

Data for Heats of formation

| Compound    |    $\Delta H^{\circ}_f$  $/ \frac{kJ}{mole}$    |    Source   |
| :------     | :------:                     | :-----      |
|  Methylcyclohexane  |  -154.8   |  https://webbook.nist.gov/cgi/inchi/InChI=1S/C7H14/c1-7-5-3-2-4-6-7/h7H,2-6H2,1H3   | 
|  Methylcyclohexane  |  -154.7   |  "Thermochemical Data of Organic Compounds", J.B. Pedley, R.D. Naylor, S.P. Kirby, *Chapman and Hall, New York*, **1986**, page 94  | 
|  1-methylcyclohexene  |  -81.25   |  https://webbook.nist.gov/cgi/inchi/InChI=1S/C7H12/c1-7-5-3-2-4-6-7/h5H,2-4,6H2,1H3   | 
|  1-methylcyclohexene  |  -43.3   |  "Thermochemical Data of Organic Compounds", J.B. Pedley, R.D. Naylor, S.P. Kirby, *Chapman and Hall, New York*, **1986**, page 93   | 
|  methylenecyclohexane  |  -25.2   | "Thermochemical Data of Organic Compounds", J.B. Pedley, R.D. Naylor, S.P. Kirby, *Chapman and Hall, New York*, **1986**, page 93  |

In [3]:
### Methylcyclohexane

measuredHf = -154.8     # Value from NIST database

CH2 = i["C-H2C2"]       # increment for CH2 groups
CH = i["C-HC3"]         # increment for CH groups
CH3 = i["C-H3C"]

print("Methylcyclohexane")
print("-----")

'''  # Commenting the whole block below out of the code. Delete the triple quotes to resore it.
print(f"The group increment for C-H2C2 is {CH2:+.2f}")
print(f"The group increment for C-HC3 is {CH:+.2f}")
print(f"The group increment for C-H3C is {CH3:+.2f}")
print("-----")
'''

Hf = 5*CH2 + 1*CH + 1*CH3

print(f"the calculated delta Hf is {Hf:+.2f} kJ/mole")

print(f"the experimental delta Hf is {measuredHf:+.2f} kJ/mole")
print("-----")

diff = measuredHf  - Hf

print (f"the strain energy is {diff:+.2f} kJ/mole")


Methylcyclohexane
-----
the calculated delta Hf is -153.61 kJ/mole
the experimental delta Hf is -154.80 kJ/mole
-----
the strain energy is -1.19 kJ/mole


In [4]:
### 1-methylcyclohexene

#measuredHf = -81.25     # Value from NIST database (same as condensed phase values - I suspect an error
measuredHf = -43.3       # Value from Pedley, Naylor & Kirby
                         # This matches with the results reported in "Strain Energy of Small Ring Hydrocarbons. 
                         # Influence of C−H Bond Dissociation Energies", Robert D. Bach and Olga Dmitrenko, 
                         # J. Am. Chem. Soc., 2004, 126, 4444-4452. https://doi.org/10.1021/ja036309a

CH3Cd = i["C-H3C"]       # increment for CH3 attached to Cd - Note: no increment in table so used 
                         # the value for CH3 attached to alkane C. According to tables in Pedley, 
                         # Naylor & Kirby, the increment for -CH3 does not change no matter what
                         # it is attached to.
 
CH2 = i["C-H2C2"]        # increment for CH2 groups in ring
CH2d = i["C-CdCH2"]      # increment for the CH2 group in the ring that is next to the double bond
CdCH = i["Cd-HC"]        # increment for the double bond carbon
CdC2 = i["Cd-C2"]        # increment for the double bond carbon that has the methyl group

print("1-methylcyclohexene")
print("-----")

'''  # Commenting the whole block below out of the code. Delete the triple quotes to resore it.
print(f"The group increment for C-H2C2 is {CH2:+.2f}")
print(f"The group increment for C-CdCH2 is {CH2d:+.2f}")
print(f"The group increment for Cd-HC is {CdCH:+.2f}")
print(f"The group increment for Cd-C2 is {CdC2:+.2f}")
print(f"The group increment for C-H3Cd is {CH3Cd:+.2f}")
print("-----")
'''

Hf = 2*CH2 + 2*CH2d + 1*CdCH + 1*CdC2 + 1*CH3Cd

print(f"the calculated delta Hf is {Hf:+.2f} kJ/mole")

print(f"the experimental delta Hf is {measuredHf:+.2f} kJ/mole")
print("-----")

diff = measuredHf - Hf

print (f"the strain energy is {diff:+.2f} kJ/mole")


1-methylcyclohexene
-----
the calculated delta Hf is -44.52 kJ/mole
the experimental delta Hf is -43.30 kJ/mole
-----
the strain energy is +1.22 kJ/mole


In [5]:
### methylenecyclohexane

measuredHf = -25.2       # Value from Pedley, Naylor & Kirby

CH2 = i["C-H2C2"]        # increment for CH2 groups in ring
CH2d = i["C-CdCH2"]      # increment for the CH2 group in the ring that is next to the double bond
CdH2 = i["Cd-H2"]        # increment for the terminal double bond carbon
CdC2 = i["Cd-C2"]        # increment for the internal double bond carbon 

print("methylenecyclohexane")
print("-----")

'''  # Commenting the whole block below out of the code. Delete the triple quotes to resore it.
print(f"The group increment for C-H2C2 is {CH2:+.2f}")
print(f"The group increment for C-CdCH2 is {CH2d:+.2f}")
print(f"The group increment for Cd-H2 is {CdH2:+.2f}")
print(f"The group increment for Cd-C2 is {CdC2:+.2f}")
print("-----")
'''

Hf = 3*CH2 + 2*CH2d + 1*CdH2 + 1*CdC2

print(f"the calculated delta Hf is {Hf:+.2f} kJ/mole")

print(f"the experimental delta Hf is {measuredHf:+.2f} kJ/mole")
print("-----")

diff = measuredHf  - Hf

print (f"the strain energy is {diff:+.2f} kJ/mole")


methylenecyclohexane
-----
the calculated delta Hf is -32.23 kJ/mole
the experimental delta Hf is -25.20 kJ/mole
-----
the strain energy is +7.03 kJ/mole


### Discussion

the three molecules investigated were...

<img src="https://raw.githubusercontent.com/blinkletter/4410PythonNotebooks/main/Class_05/images/cyclohexanes.png" alt="cyclohexanes" width="500" class="centre">

The ring strain is found to be as follows...

| Compound    |    Strain Energy  $/ \frac{kJ}{mole}$    | 
| :------     | :------:                     | 
|  Methylcyclohexane  |  -1.19   |   
|  1-Methylcyclohexene  |  +1.22   |  
|  Methylenecyclohexane  |  +7.03   | 

The methylcyclohexane and 1-methylcyclohexene had negligible ring strain. The exocyclic double bond in methylenecyclohexane resulted in the moderate value of $+7.0 \frac{kJ}{mole}$ in ring strain. 

### The Journey

Along the way we found that the NIST database did not contain gas-phase heats of formation for methylenecyclohexane and the value reported for 1-methylcyclohexene was incorrect. We found the correct values for these two compounds in Pedley, Naylor & Kirby (see references section at end of this notebook) 

### Looking for Confirmation
Was I correct? I searched the literature for values for strain energy in these compounds and came across the following contribution...

"Strain Energy of Small Ring Hydrocarbons. Influence of C−H Bond Dissociation Energies", Robert D. Bach and Olga Dmitrenko, *J. Am. Chem. Soc.*, **2004**, *126*, 4444-4452. https://doi.org/10.1021/ja036309a

It it were reported enthalpies of hydrogenation for the two alkenes. Both reactions give the methylcyclohexane as product. Compare this analysis to that for substituted alkene on page 113 (figure 2.17) of your textbook.

| Compound    | Heat of hydrogenation $\Delta H^{\circ}_{hyd}$  $/ \frac{kJ}{mole}$  | 
| :------    | :------------------:   | 
|  Methylcyclohexane      |  -        |   
|  1-Methylcyclohexene    |  -106.6   |  
|  Methylenecyclohexane   |  -122.9   | 
|  cyclohexene            |  -116.6   | 

The release of energy as the double bond is hydrogenated appears to be about $16~\frac{kJ}{mole}$ greater for methylenecyclohexane compared to 1-Methylcyclohexene. However we can attribute approximately $10~\frac{kJ}{mole}$ of this effect due to the fact that 1-Methylcyclohexene is a triply substitued double bond compared to methylenecyclohexane being double substituted ($-116.6$ vs $-106.6~\frac{kJ}{mole}$. So we could estimate the difference in strain energy to be about $6~\frac{kJ}{mole}$.

that is in strong agreement with the strain energies obstianed via group increments and experimental heats of formation. the difference there was also about $6~\frac{kJ}{mole}$ ($+1.22$ vs $+7.03~\frac{kJ}{mole}$)



## References
Below are some places that I found data used in this exercise. If you find useful tables or databases please share with the class.
<hr>

Heats of combustion from CRC handbook: 
https://hbcp-chemnetbase-com.proxy.library.upei.ca/faces/documents/05_05/05_05_0001.xhtml

Heats of Formation from Argonne National laboratory: 
https://atct.anl.gov/Thermochemical%20Data/version%201.122r/index.php

NIST Webbook: 
https://webbook.nist.gov/chemistry/

See also : "Chemical calculations ; an introduction to the use of mathematics in chemistry", Sidney W. Benson, *Wiley \& Sons*, **1971**, UPEI library Stacks (2nd floor) QD42.B453 1971

and

"[Thermochemical Data of Organic Compounds](https://moodle31.upei.ca/mod/resource/view.php?id=759268)", J.B. Pedley, R.D. Naylor, S.P. Kirby, *Chapman and Hall, New York*, **1986**, 1-792 (pirated from the web - honestly, the only legitimate copy that I could find within a library system was in Australia). - There was one error/omission in table 2.4 that I corrected using data from this book.

"Organic Compounds – Physical and Thermochemical Data".  J.G. Speight, *Rules of Thumb for Petroleum Engineers, John Wiley & Sons*, **2017**. 539-545.  https://ebookcentral.proquest.com/lib/upei/detail.action?docID=4812517
